### ISSUE TITLE
unique in special axis

### ISSUE THREAD

**Issue by meijieru** [tensorflow/tensorflow#11575]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 14.04
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
1.2
- **Python version**: 
3.6
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
cuDNN v5.1
- **GPU model and memory**:
pascal titan x
- **Exact command to reproduce**:
None

**Feature Request**
For getting unique row, we can use numpy's `unique` by specifying the axis. However, in tensorflow, `unique` only support 1D tensor, which is not convenient.

**Comment by drpngx** [comment]
@aselle WDYT?

**Comment by BlGene** [comment]
@aselle I think this would be useful, at the moment I have to do this:

```
assert_max = tf.Assert(tf.reduce_all(indices < 256), [indices])
with tf.control_dependencies([assert_max]):
  indices = tf.identity(indices)
indices = tf.cast(indices, tf.uint8)

# find unique indices, because SparseTensor needs this
indices_pad = tf.pad(indices,[[0,0],[0,1]])  # shape to (N, 4)
assert(indices_pad.get_shape()[1].value == 4)
indices_bc  = tf.bitcast(indices_pad, tf.int32)
unique32, _ = tf.unique(indices_bc)
indices = tf.cast(tf.bitcast(unique32, tf.uint8)[:,:3],tf.int64)
```

**Comment by yongtang** [comment]
Added a PR #12952 for the support of `axis` for `tf.unique`.

**Comment by nbro** [comment]
This feature is still not supported (at least in the Python API). An easy workaround is described in this answer: https://stackoverflow.com/a/60046022/3924118.

**Comment by zaqqwerty** [comment]
See #15644, this is currently not a feature in tf 2.2

**Comment by shariq-audiofocus** [comment]
@zaqqwerty  - one workaround (no gradients available) - 
```
x.shape  # (100, 2)
unique_results = tf.raw_ops.UniqueV2(x=not_unique, axis=0).y
unique_results.shape  # (89, 2)
```

See here: 
https://www.tensorflow.org/api_docs/python/tf/raw_ops
https://www.tensorflow.org/api_docs/python/tf/raw_ops/UniqueV2
